python/GafferDelightUI/ShaderMenu.py

```diff
@@ -71,35 +71,158 @@ def __shaderSubMenu( oslPath ) :
 			for m in q.metadata :
 				metadata[m.name] = m.value
 
-			shaderName = metadata.get( "niceName", IECore.CamelCase.toSpaced( shader ) )
+			shaderFileName = metadata.get( "niceName", shader )
 
+			shaderName = metadata.get( "niceName", IECore.CamelCase.toSpaced( shader ) )
 			shaderName = shaderName.replace( "&", "and" )
 			shaderName = shaderName[2:].lstrip() if shaderName.lower().startswith( "dl" ) else shaderName
 
 			if "hidden" in metadata.get( "tags", [] ) :
 				continue
+			if "houdini" in metadata.get( "tags", [] ) :
+				continue
 
 			subMenu = None
 			for tag in metadata.get( "tags", [] ) :
-				if tag == "surface" and shader.startswith( "dl" ) :
-					subMenu = "Surface"
-				elif tag == "volume" :
-					subMenu = "Volume"
-				elif tag == "displacement" and shader.startswith( "dl" ) :
-					subMenu = "Displacement"
-				elif tag == "texture/2d" :
-					subMenu = "Texture 2D"
-				elif tag == "texture/3d" :
-					subMenu = "Texture 3D"
-				elif tag == "utility" :
-					subMenu = "Utility"
-				elif tag == "toon" :
-					subMenu = "Toon"
-				else :
-					subMenu = "Other"
+				subMenu = tag.capitalize()
 
-			if subMenu is None :
+			if shaderFileName == "distantLight" :
+				continue
+			elif shaderFileName == "directionalLight" :
+				continue
+			elif shaderFileName == "distantLight" :
+				continue
+			elif shaderFileName == "pointLight" :
+				continue
+			elif shaderFileName == "spotLight" :
+				continue
+			elif shaderFileName == "environmentLight" :
+				continue
+			elif shaderFileName == "displacementShader" :
+				continue
+			elif shaderFileName == "surfaceShader" :
+				continue
+			elif shaderFileName == "plusMinusAverage" :
+				continue
+			elif shaderFileName == "addDoubleLinear" :
 				continue
+			elif shaderFileName == "multDoubleLinear" :
+				continue
+			elif shaderFileName == "tripleShadingSwitch" :
+				continue
+			elif shaderFileName == "pfx" :
+				continue
+			elif shaderFileName == "particleCloud" :
+				continue
+			elif shaderFileName == "shadingEngine_surface" :
+				continue
+			elif shaderFileName == "surfaceLuminance" :
+				continue
+			elif shaderFileName == "filter_multiply" :
+				continue
+			elif shaderFileName == "lens_distortion" :
+				continue
+			elif shaderFileName == "dl3DelightMaterial" :
+				continue
+			elif shaderFileName == "transparent" :
+				continue
```

Comment:
Small nitpick if you feel like changing it, it might be a bit more readable to do something like
```
if shaderFileName in [
    "distantLight",
    "directionalLight",
    "pointLight",
    ...
] :
    continue
```
instead of the repeated `elif`.

Reply:
That code looks much better. Thanks for the suggestion and fixed in the latest commit.